### ISSUE TITLE
Legends: Surface datalinks in viz legends

### ISSUE THREAD

**Issue by Fangkey** [grafana/grafana#89384]
**Why is this needed**:

We can add datalink in pie chart, then when we click on one item of the pie chart, we can set a global filter in URL and do a full page drilldown refresh. This is a base feature for modern dashboards. But when an item has very percentage in the pie chart, let say <1%, it is very difficult to click on it. Instead, we can click on the item in the legend to do the same drilldown action, this is one of the meanings of legend for dashboard. Most dashboard platforms support this feature, but for Grafana, I see there was the same request from 10 years ago but be ignored. I think it is just a pure frontend UI improvement, since we already have datalink for pie chart itself, what we need to do is to call the same handler when click on legend item.

In following pie chart, we have several small items, they are smaller than 1 pixel in screen, it is a very bad user experience to select among them:
![image](https://github.com/grafana/grafana/assets/5007415/80845df0-3c12-4b42-a5a6-fbfb08cfc72c)

**What would you like to be added**:

Enable interactive datalink click for legend of visualization, at least for pie chart.

**Who is this feature for?**

Nearly all modern dashboard platforms support this feature, it should be a standard solution needed by everyone.

**Comment by leeoniya** [comment]
good idea, but we have to figure out what to do about the existing click functionality that toggles series on/off (in all panels, not just pie chart)
